tools/README.md

```diff
@@ -0,0 +1,90 @@
+# `mongo-c-tools`
+
+This directory contains shared scripts and tools for native C and C++
+development.
+
+
+## As a Subrepo
+
+If this directory contains a `.gitrepo` file, then this directory is being
+tracked as a *subrepo* of some parent repository. If this file does not exist
+but a `.git/` directory *does* exist, then you are looking at the tools
+repository itself.
+
+[`git subrepo`](https://github.com/ingydotnet/git-subrepo) is a set of scripts
+and conventions for pushing/pulling/syncing changes in a subdirectory between
+different repositories with full history tracking. *Using* the contents of this
+directory does not require the `git subrepo` tool, but pushing and pulling
+changes in this subrepo directory will require `git subrepo` (or knowing
+the appropriate commands to execute for the same effect).
+
+If you are simply using these scripts, you do not have to deal with the subrepo
+commands. If you wish to modify/sync these scripts, read on below.
+
+
+### Preparing
+
+`git-subrepo` tracks a subrepo directory using a remote+branch pair as the
+"source of truth" for the subrepo contents. The remote+branch pair is fixed in
+the `.gitrepo` file. In this subrepo directory, the `remote` is not set to a
+URL, but to a name `tools-origin`. In order to push/pull, you will need to
+ensure that you have a `tools-origin` set to the proper URL. Unless you "know
+what you are doing", you will want to use the upstream `mongo-c-driver`
+repository as the remote:
```

Comment:
For rationale documentation purposes, why `mongodb/mongo-c-driver` as the remote rather than a separate `mongodb(-labs)/mongo-c-driver-tools` repo (a la `mongodb-labs/drivers-evergreen-tools`)? I find the use of the "disjoint branch" to be strange. It is effectively already behaving as a separate git repo that is using an [arbitrary commit](https://github.com/mongodb/mongo-c-driver/commit/9aa60260db9d9c45a0ae3c5bc65e2adbba1a5879) as a common ancestor with the main branch. If we want to use the "disjoint branch" method, might as well go all-in and remove the common ancestor entirely (we won't ever be merging the `mongo-c-tools` branch itself into the main branch anyways, right?).

Reply:
No reason at all other than "I didn't bother to create a new repo for it" :). At `$prior_job` we used a separate repo for the subrepo target, and it works great with that. It'd probably actually be better, since there wouldn't be need to juggle the remote URL.

Reply:
Oh also I don't have the permission to create a new repo in the org, so I just used what I had available.

Reply:
We already have a project at `10gen/mongo-c-driver-tools.git` which might be a good candidate, assuming that a project which isn't public is acceptable in this case.

Reply:
That could very well work. It does not need to be a publicly accessible repo.

Reply:
I support use of `10gen/mongo-c-driver-tools`. AFAIK the only actively used script in the repo is `release.py`.